script.js

```diff
@@ -0,0 +1,208 @@
+      cancelButton.onclick = () => cancelFun(index);
+      container.appendChild(cancelButton);
+
+      const imageAccept = document.createElement("img");
+      imageAccept.src = "img/accept.svg";
+      acceptButton.appendChild(imageAccept);
+
+      const imageCancel = document.createElement("img");
+      imageCancel.src = "img/cancel.svg";
+      cancelButton.appendChild(imageCancel);
+    } else {
+      let text = document.createElement("p");
+      text.innerText = `${index + 1}) Магазин: "${item.place}"`;
+      
+      let date = document.createElement('p');
+      date.innerText = new Date().toLocaleDateString();
+      
+      let spent = document.createElement("p");
+      spent.innerText = listCosts[index].spent;
+      
+      sum = sum + newSpent;
+      summ.innerText = `Итого: ${sum} Р.`;
+      
+      const buttonArea = document.createElement("div");
+      buttonArea.className = "buttonArea";
+      
+      const changeButton = document.createElement("button");
+      changeButton.onclick = () => changeFun(index);
+      
+      const imageEdit = document.createElement("img");
+      imageEdit.src = "img/editor.svg";
+      changeButton.appendChild(imageEdit);
+      
+      const deleteButton = document.createElement("button");
+      deleteButton.onclick = () => deleteFun(_id);
+      
+      const imageDelete = document.createElement("img");
+      imageDelete.src = "img/remove.svg";
+      deleteButton.appendChild(imageDelete);
+      
+      buttonArea.appendChild(changeButton);
+      buttonArea.appendChild(deleteButton);
+      container.appendChild(text);
+      container.appendChild(date);
+      container.appendChild(spent);
+      container.appendChild(buttonArea);
+    };
+    
+    content.appendChild(container);
+  });
+};
+
+let inputPlaceValue = (event) => {
+  inputPlace.value = event.target.value;
+};
+
+let inputSpentValue = (event) => {
+  inputSpent.value = event.target.value;
+};
+
+let inputChangePlaceValue = (event) => {
+  inputPlaceChange.value = event.target.value;
+};
+
+let inputChangeSpentValue = (event) => {
+  inputSpentChange.value = event.target.value;
+};
+
+const deleteFun = async (_id) => {
+  const resp = await fetch(`http://localhost:8000/deleteCosts?_id=${_id}`, {
+    method: "DELETE",
+  });
+  let result = await resp.json();
+  listCosts = result.data;
+  
+  render();
+};
+
+const changeFun =  (index) => {
+  listCosts[index].flag = 1;
+  render();
+};
+
+const cancelFun = (index) => {
+  listCosts[index].flag = 0;
+
+  render();
+};
+
+const acceptFun = async (place, spent, _id, index) => { 
+  place = inputPlaceChange.value;
+  spent = inputSpentChange.value;
+  const resp = await fetch("http://localhost:8000/changeCost", {
+    method: "PATCH",
+    headers: {
+      "Content-Type": "application/json; charset=utf-8",
+      "Access-Control-Allow-Origin": "*",
+    },
+    body: JSON.stringify({
+      place: place,
+      date: new Date(),
+      spent: spent,
+      _id: _id
```

Comment:
only place, spent, _id.  if u have the same key and value in an object, write only the key